src/metal.cs

```diff
@@ -4164,24 +4571,585 @@ interface MTLCounterSampleBuffer {
+
+		[Export ("sampleBufferAttachments")]
+		MTLBlitPassSampleBufferAttachmentDescriptorArray SampleBufferAttachments { get; }
+	}
+
+	[Mac (11,0), iOS (14,0), TV (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface MTLBlitPassSampleBufferAttachmentDescriptor : NSCopying {
+		[NullAllowed, Export ("sampleBuffer", ArgumentSemantic.Retain)]
+		IMTLCounterSampleBuffer SampleBuffer { get; set; }
+
+		[Export ("startOfEncoderSampleIndex")]
+		nuint StartOfEncoderSampleIndex { get; set; }
+
+		[Export ("endOfEncoderSampleIndex")]
+		nuint EndOfEncoderSampleIndex { get; set; }
+	}
+
+	[Mac (11,0), iOS (14,0), TV (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface MTLBlitPassSampleBufferAttachmentDescriptorArray {
+		[Export ("objectAtIndexedSubscript:")]
+		MTLBlitPassSampleBufferAttachmentDescriptor GetObject (nuint attachmentIndex);
+
+		[Export ("setObject:atIndexedSubscript:")]
+		void SetObject ([NullAllowed] MTLBlitPassSampleBufferAttachmentDescriptor attachment, nuint attachmentIndex);
+	}
+
+	[Mac (11,0), iOS (14,0), TV (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface MTLCommandBufferDescriptor : NSCopying {
+
+		[Field ("MTLCommandBufferEncoderInfoErrorKey")]
+		NSString BufferEncoderInfoErrorKey { get; }
+
+		[Export ("retainedReferences")]
+		bool RetainedReferences { get; set; }
+
+		[Export ("errorOptions", ArgumentSemantic.Assign)]
+		MTLCommandBufferErrorOption ErrorOptions { get; set; }
+	}
+
+	[Mac (11,0), iOS (14,0), TV (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface MTLComputePassDescriptor : NSCopying {
+		[Static]
+		[Export ("computePassDescriptor")]
+		MTLComputePassDescriptor Create ();
+
+		[Export ("dispatchType", ArgumentSemantic.Assign)]
+		MTLDispatchType DispatchType { get; set; }
+
+		[Export ("sampleBufferAttachments")]
+		MTLComputePassSampleBufferAttachmentDescriptorArray SampleBufferAttachments { get; }
+	}
+
+	[Mac (11,0), iOS (14,0), TV (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface MTLComputePassSampleBufferAttachmentDescriptor : NSCopying {
+
+		[NullAllowed, Export ("sampleBuffer", ArgumentSemantic.Retain)]
+		IMTLCounterSampleBuffer SampleBuffer { get; set; }
+
+		[Export ("startOfEncoderSampleIndex")]
+		nuint StartOfEncoderSampleIndex { get; set; }
+
+		[Export ("endOfEncoderSampleIndex")]
+		nuint EndOfEncoderSampleIndex { get; set; }
+	}
+
+	[Mac (11,0), iOS (14,0), TV (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface MTLComputePassSampleBufferAttachmentDescriptorArray {
+		[Export ("objectAtIndexedSubscript:")]
+		MTLComputePassSampleBufferAttachmentDescriptor GetObject (nuint attachmentIndex);
+
+		[Export ("setObject:atIndexedSubscript:")]
+		void SetObject ([NullAllowed] MTLComputePassSampleBufferAttachmentDescriptor attachment, nuint attachmentIndex);
+	}
+
+	[Mac (11,0), iOS (14,0), TV (14,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject))]
+	interface MTLFunctionDescriptor : NSCopying {
+		[Static]
+		[Export ("functionDescriptor")]
+		MTLFunctionDescriptor Create { get; }
```

Comment:
```suggestion
		MTLFunctionDescriptor Create ();
```

Also you might need to inline this static `Create ()` function to subclasses of `MTLFunctionDescriptor` that actually return the subclass type like `MTLIntersectionFunctionDescriptor`